src/test/java/com/google/cloud/spanner/pgadapter/JdbcMockServerTest.java

```diff
@@ -443,12 +494,14 @@ public void testTwoDmlStatements() throws SQLException {
     }
 
     // The DML statements are split by the JDBC driver and sent as separate statements to PgAdapter.
-    // PgAdapter therefore also simply executes these as separate DML statements.
-    assertTrue(mockSpanner.getRequestsOfType(ExecuteBatchDmlRequest.class).isEmpty());
-    List<ExecuteSqlRequest> requests = mockSpanner.getRequestsOfType(ExecuteSqlRequest.class);
-    assertEquals(2, requests.size());
-    assertEquals(INSERT_STATEMENT.getSql(), requests.get(0).getSql());
-    assertEquals(UPDATE_STATEMENT.getSql(), requests.get(1).getSql());
+    // The Sync message is however sent after the second DML statement, which means that PGAdapter
+    // is able to batch these together into one ExecuteBatchDml statement.
+    assertEquals(1, mockSpanner.countRequestsOfType(ExecuteBatchDmlRequest.class));
```

Comment:
Here is the proof that batching in extended query protocol now uses `ExecuteBatchDml` when that is possible.